backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/codec/impl/HibernateSearchKnnVectorsFormat.java

```diff
@@ -20,7 +20,10 @@
 import org.apache.lucene.index.SegmentWriteState;
 
 public class HibernateSearchKnnVectorsFormat extends KnnVectorsFormat {
-	public static final int DEFAULT_MAX_DIMENSIONS = KnnVectorsFormat.DEFAULT_MAX_DIMENSIONS;
+	// OpenSearch has a limit of 16000
+	// Elasticsearch has a limit of 4096
+	// We'll keep it at 4096 for now as well:
+	public static final int DEFAULT_MAX_DIMENSIONS = 4096;
```

Comment:
Isn't there any documentation (asciidoc/javadoc) to update as a result of this bump? We didn't use to document the max?

Reply:
Thanks for pointing that out! updated and added links to ES/OS docs so we don't need to track  any changes on their side.

it's a good thing that I've left those `TODO: HSEARCH-5020` todos in docs and code so whenever we come to work on this ticket, we don't miss any places to update... and then just ignore them 🙈 😭 🤦🏻‍♂️